email_app/app/utils/app_utils.js

```diff
@@ -65,15 +64,37 @@ export const deserialiseArray = (str) => {
   return Uint8Array.from(arrItems);
 }
 
-export const genKeyPair = () => {
-  let {keyType, privateKey, publicKey} = sodium.crypto_box_keypair('hex');
-  return {privateKey, publicKey};
+export const genKeyPair = (app) => {
+  let rawKeyPair = {};
+  return app.crypto.generateEncKeyPair()
+  .then(keyPair => keyPair.pubEncKey.getRaw()
+    .then(rawPubEncKey => {
+      rawKeyPair.publicKey = rawPubEncKey;
+      return;
+    })
+    .then(() => keyPair.secEncKey.getRaw())
+    .then(rawSecEncKey => {
+      rawKeyPair.privateKey = rawSecEncKey;
+      return rawKeyPair;
+    })
+  )
 }
 
-export const encrypt = (input, pk) => sodium.crypto_box_seal(input, Buffer.from(pk, 'hex'), 'hex');
+export const encrypt = (app, input, pk) => {
+  console.log("public key used for encrypt: ", new Uint8Array(pk.buf));
 
-export const decrypt = (cipherMsg, sk, pk) => sodium.crypto_box_seal_open(
-                              Buffer.from(cipherMsg, 'hex'),
-                              Buffer.from(pk, 'hex'),
-                              Buffer.from(sk, 'hex'),
-                              'text');
+  return app.crypto.pubEncKeyKeyFromRaw(new Uint8Array(pk.buf))
```

Comment:
Perhaps the serialisation issue is due to the use of `Uint8Array`...? have you tried with creating directly a `Buffer` instead, or just passing the string `pk.buf.toString()`?